library/std/src/sys/sgx/abi/usercalls/alloc.rs

```diff
@@ -304,6 +305,100 @@ where
     }
 }
 
+/// Copies `len` bytes of data from enclave pointer `src` to userspace `dst`
+///
+/// This function mitigates stale data vulnerabilities
+/// https://www.intel.com/content/www/us/en/security-center/advisory/intel-sa-00615.html
```

Comment:
This link makes it a little clearer that the vulnerability is specifically about partial writes ("Any write that is smaller than 8 bytes or is not 8-byte aligned."), which explains why we're only looking at `dst` alignment, and why the aligned chunks remain so simple. Could we let the aligned part go back to a normal `ptr::copy`?

https://www.intel.com/content/www/us/en/developer/articles/technical/software-security-guidance/technical-documentation/processor-mmio-stale-data-vulnerabilities.html

Reply:
> Could we let the aligned part go back to a normal ptr::copy?

Actually, I guess that's a bad idea because we need to **guarantee** that all of those writes are full 8-bytes at a time.

Reply:
Agreed! I've extended the documentation and added the reference